scripted-diff: Replace CCriticalSection with RecursiveMutex

maflcko opened the pull request:
`RecursiveMutex` better clarifies that the mutex is recursive, see also the standard library naming: https://en.cppreference.com/w/cpp/thread/recursive_mutex

For that reason, and to avoid different people asking me the same question repeatedly (e.g. https://github.com/bitcoin/bitcoin/pull/15932#pullrequestreview-339175124 ), remove the outdated alias `CCriticalSection` with a scripted-diff

Commit:
scripted-diff: Replace CCriticalSection with RecursiveMutex

-BEGIN VERIFY SCRIPT-
 # Delete outdated alias for RecursiveMutex
 sed -i -e '/CCriticalSection/d'                 ./src/sync.h
 # Replace use of outdated alias with RecursiveMutex
 sed -i -e 's/CCriticalSection/RecursiveMutex/g' $(git grep -l CCriticalSection)
-END VERIFY SCRIPT-

practicalswift commented:
Concept ACK: explicit is better than implicit :)

promag reviewed (commented):
ACK dc0e8b6b3f665cf764af5c26e3745316ceb95302.

practicalswift commented:
ACK dc0e8b6b3f665cf764af5c26e3745316ceb95302 -- scripted diff looks correct

jtimon commented:
ACK dc0e8b6b3f665cf764af5c26e3745316ceb95302

Empact commented:
ACK https://github.com/bitcoin/bitcoin/pull/17891/commits/dc0e8b6b3f665cf764af5c26e3745316ceb95302

fanquake commented:
Ugh. Messed up my merge order here, should have done this before #17906. @MarcoFalke can you please rebase.

Commit:
scripted-diff: Bump copyright of files changed in 2020

-BEGIN VERIFY SCRIPT-
./contrib/devtools/copyright_header.py update ./
-END VERIFY SCRIPT-

MarcoFalke commented:
Rebased and added another scripted diff to also bump the copyright headers of modified files in one go

practicalswift commented:
ACK e09c701e0110350f78366fb837308c086b6503c0 -- scripted diff looks correct

promag commented:
ACK e09c701e0110350f78366fb837308c086b6503c0
